sys/kern/devfs.c

```diff
@@ -149,9 +149,12 @@ static int devfs_fop_write(file_t *fp, uio_t *uio) {
 }
 
 static int devfs_fop_close(file_t *fp) {
+  int error;
   devnode_t *dev = fp->f_data;
   refcnt_release(&dev->refcnt);
-  return dev->ops->d_close(dev, fp);
+  error = dev->ops->d_close(dev, fp);
+  vnode_drop(fp->f_vnode);
+  return error;
```

Comment:
Why not return `dev->ops->d_close(dev, fp)` since it is a value of `error`?
Truth be said, now I know that the value of sophisticated expression with two arrows has a value of its error.

Reply:
We need to drop a reference before we return `error`.